Sources/ObservableStore/ObservableStore.swift

```diff
@@ -352,14 +352,20 @@ where Model: ModelProtocol
     /// `send(_:)` is run *synchronously*. It is up to you to guarantee it is
     /// run on main thread when SwiftUI is being used.
     public func send(_ action: Model.Action) {
+        DispatchQueue.main.async { @MainActor in
```

Comment:
Is there a reason we use this instead of the `Task` interface?

`DispatchQueue` is part of the old Grand Central Dispatch async API and does not coordinate with Swift's concurrency green thread implementation. Best avoided.

Reply:
I'm having trouble getting the tests to consistently pass using `Task { @MainActor in }`, but on further experimentation this seems to be an issue regardless of approach. 

Each test passes individually but running them together has them all competing to schedule tasks. Let's chat about the best approach on a call.